apps/wing/src/commands/compile.ts

```diff
@@ -107,6 +107,19 @@ export async function compile(entrypoint: string, options: ICompileOptions) {
     },
     __dirname: workDir,
     __filename: artifactPath,
+
+    // since the SDK is loaded in the outer VM, we need these to be the same class instance,
+    // otherwise "instanceof" won't work between preflight code and the SDK. this is needed e.g. in
+    // `serializeImmutableData` which has special cases for serializing these types.
```

Comment:
Would it be better to solve this by not using `require.resolve` above to preload wingsdk? I think we could instead override `require` in this VM to resolve from the current dir.

We may need to do something like that anyways once we have more complicated deps.

Reply:
That's a good idea. Since this can be a bit messy I'll do it in a subsequent commit.

Reply:
https://github.com/winglang/wing/issues/1148